### PR TITLE
sync_blocks cleanup

### PR DESCRIPTION
**Pull request by ryanofsky**
This PR is a followup to #9058. It adds more detail to sync_block exceptions and fixes the buggy sync_blocks timeout implementation.

### CODE REVIEW

**Comment by morcos** [comment]
tested ACK

**Comment by MarcoFalke** [comment]
Thanks!

utACK 9683a332e8565be7399b17bf77b9fddd432d7a20

File: qa/rpc-tests/test_framework/util.py [hunk 1]
```diff
@@ -123,19 +123,26 @@ def str_to_b64str(string):
...
-        if heights == [heights[0]] * len(heights):
-            raise AssertionError("Block sync failed: (Hashes don't match)")
-        timeout -= wait
+        if all(h == heights[0] for h in heights):
+            raise AssertionError("Block sync failed, mismatched block hashes:{}".format(
+                                 "".join("\n  {!r}".format(tip) for tip in tips)))
         maxheight = max(heights)
-    raise AssertionError("Block sync failed with heights: {}".format(heights))
+        cur_time = time.time()
+    raise AssertionError("Block sync to height {} timed out{}:".format(
```
Comment (MarcoFalke) [inline qa/rpc-tests/test_framework/util.py]: µNit: I think you want the colon in front of the format specifier?

Comment (ryanofsky) [inline qa/rpc-tests/test_framework/util.py]: Thanks, fixed.

**Pull request merged**